tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_SphericalCompression.cpp

```diff
@@ -0,0 +1,595 @@
+      [&center, &map, &time, &functions_of_time](
+          const std::array<double, 3>& orig_point_1,
+          const std::array<double, 3>& orig_point_2,
+          const bool check_if_equal) {
+        const std::array<double, 3> mapped_point_1{
+            map(orig_point_1, time, functions_of_time)};
+        const std::array<double, 3> mapped_point_2{
+            map(orig_point_2, time, functions_of_time)};
+        const std::array<double, 3> scale_factor_1{
+            {(orig_point_1[0] - mapped_point_1[0]) /
+                 (orig_point_1[0] - center[0]),
+             (orig_point_1[1] - mapped_point_1[1]) /
+                 (orig_point_1[1] - center[1]),
+             (orig_point_1[2] - mapped_point_1[2]) /
+                 (orig_point_1[2] - center[2])}};
+        const std::array<double, 3> scale_factor_2{
+            {(orig_point_2[0] - mapped_point_2[0]) /
+                 (orig_point_2[0] - center[0]),
+             (orig_point_2[1] - mapped_point_2[1]) /
+                 (orig_point_2[1] - center[1]),
+             (orig_point_2[2] - mapped_point_2[2]) /
+                 (orig_point_2[2] - center[2])}};
+        Approx custom_approx = Approx::custom().epsilon(1.0e-9).scale(1.0);
+        if (check_if_equal) {
+          CHECK(scale_factor_1[0] == custom_approx(scale_factor_2[0]));
+          CHECK(scale_factor_1[1] == custom_approx(scale_factor_2[1]));
+          CHECK(scale_factor_1[2] == custom_approx(scale_factor_2[2]));
+        } else {
+          CHECK(scale_factor_1[0] != custom_approx(scale_factor_2[0]));
+          CHECK(scale_factor_1[1] != custom_approx(scale_factor_2[1]));
+          CHECK(scale_factor_1[2] != custom_approx(scale_factor_2[2]));
+        }
+      };
+
+  // Set up distributions for choosing radii in the three regions:
+  // smaller than min_radius, larger than max_radius, and in between
+  constexpr double eps{0.01};
+
+  double lower_radius = min_radius + eps;
+  double upper_radius = max_radius - eps;
+  if constexpr (InteriorMap) {
+    lower_radius = eps;
+    upper_radius = min_radius - eps;
+  }
+  std::uniform_real_distribution<> radius_dis{lower_radius, upper_radius};
+  if constexpr (InteriorMap) {
+    // In the interior, two random points should change radius by the same
+    // factor, and that factor is the same as for a point exactly at
+    // the minimum radius
+    check_point_scale_factors(point(radius_dis(*generator)),
+                              point(radius_dis(*generator)), true);
+    check_point_scale_factors(point(radius_dis(*generator)), point(min_radius),
+                              true);
+  } else {
+    // In the middle, two points should only move radially, but by
+    // a different amount
+    check_point_scale_factors(point(radius_dis(*generator)),
+                              point(radius_dis(*generator)), false);
+  }
+
+  // Check that a point is invertible with the supplied functions_of_time
+  CHECK(static_cast<bool>(
```

Comment:
for `std::optional`, we can use `.has_value()` which is more readable (and below)